docs/tutorials/write-your-first-comit-app/04-2-taker-takes-order.md

```diff
@@ -0,0 +1,168 @@
+import { formatEther } from "ethers/utils";
+
+(async function main() {
+    console.log("COMIT Taker app");
+
+    const configPath = path.join(os.homedir(), ".create-comit-app", "env");
+    dotenv.config({path: configPath});
+
+    const bitcoinWallet = await InMemoryBitcoinWallet.newInstance(
+        "regtest",
+        process.env.BITCOIN_P2P_URI!,
+        process.env[`BITCOIN_HD_KEY_${1}`]!
+    );
+    // Waiting for the Bitcoin wallet to read the balance
+    await new Promise(r => setTimeout(r, 1000));
+
+    const ethereumWallet = new EthereumWallet(
+        process.env.ETHEREUM_NODE_HTTP_URL!,
+        process.env[`ETHEREUM_KEY_${1}`]!
+    );
+
+    let taker = await createActor(
+        bitcoinWallet,
+        ethereumWallet,
+        process.env[`HTTP_URL_CND_${1}`]!
+    );
+
+    console.log(
+        "[Taker] Bitcoin balance: %f, Ether balance: %f",
+        (await taker.bitcoinWallet.getBalance()).toFixed(2),
+        parseFloat(
+            formatEther(await taker.ethereumWallet.getBalance())
+        ).toFixed(2)
+    );
+
+    const makerNegotiatorUrl = "http://localhost:2318/";
+
+    const takerNegotiator = new TakerNegotiator(
+        taker.comitClient,
+        makerNegotiatorUrl
+    );
+
+    const criteria = {
+        buy: {
+            ledger: "bitcoin",
+            asset: "bitcoin",
+            minNominalAmount: "1",
+        },
+        sell: {
+            ledger: "ethereum",
+            asset: "ether",
+        },
+        minRate: 0.001,
+    };
+
+    const order = await takerNegotiator.getOrder(criteria);
+
+    console.log("Rate offered: ", order.getOfferedRate().toString());
+
+    const swap = await order.take();
+
+    console.log(
+        `Took the following order: %s:%s for a rate of %d:%d`,
+        order.rawOrder.ask.asset,
+        order.rawOrder.bid.asset,
+        order.rawOrder.ask.nominalAmount,
+        order.rawOrder.bid.nominalAmount
+    );
+
+    process.exit();
+})();
+```
+
+Note, that in order to properly retrieve an order at the taker side, the maker application has to run at this stage!
+If your maker app is not running you can start it with `yarn maker`.
+
+Ensure that your maker app is running and then start the taker app with `yarn taker` - it should print:
+```
+yarn run v1.22.0
+$ ts-node ./src/taker.ts
+COMIT Taker app
+[Taker] Bitcoin balance: 10, Ether balance: 1000
+Rate offered:  0.02
+Took the following order: ether:bitcoin for a rate of 50:1
+✨  Done in 2.78s.
+```
+
+Once you run the taker app, the **maker app** will terminate as well:
+```
+yarn run v1.22.0
+$ ts-node ./src/maker.ts
+COMIT Maker app
+[Maker] Bitcoin balance: 10, Ether balance: 1000
+Maker's Negotiation Service is listening on localhost:2318.
+Waiting for someone to take my order at: http://127.0.0.1:2318
+✨  Done in 11.01s.
+```
+
+This is, because the maker was waiting for the incoming swap request on cnd and has received it from the taker.
```

Comment:
@D4nte @bonomat With the current split in negotiation steps and execution steps we run into the  problem, that if the taker at this stage takes the order, an actual swap request is triggered through cnd. 
Both applications then terminate (because the execution was not added yet).
But the swap request of course still exists in cnd. 

Given the "simplified" swap-listening mechanism on the maker side, we always chose the first swap. 
On the taker side, however, we will always step through fetching an order and then initialising the swap from taking that order.
This means that after this stage the remaining swap in cnd will cause trouble.

I cannot just omit that detail, otherwise devs might run into that problem and get confused.
How shall I deal with this? I see three possible solutions:

1. Should we add matching logic, so it does not pick "the first" swap in the list?
2. Should we just let the tutorial user restart the environment?
3. Should we combine the "Taker takes order" section with the execution for the Taker? (probably the easiest, but I feel the separation between negotiation and execution is not that nice then.)

Reply:
I'm not sure if I follow your problem. What do you mean with?

> Both applications then terminate (because the execution was not added yet).

Provided that this example is run one once, then I think everything is fine as 1 order is taken from the taker and then all other steps are executed. There is only 1 swap available on both sides at any time.

Reply:
- Let the dev restart at the environment at this stage.
- Open an issue to fix that, this is a sign that we should provide a better API. The Maker only accepts the swap that the taker took. The Maker should know what swap he accepted via the negotiation interface and we should be manipulating this swap (`swapHandle`) instead of grabbing the first ongoing swap returned by cnd.